### ISSUE TITLE
enable_mixed_precision_graph_rewrite error

### ISSUE THREAD

**Issue by sjtusmartboy** [tensorflow/tensorflow#37666]
tensorflow  1.14 and ubuntu 18.04
 
I'm trying the mixed precision training using the `tf.train.experimental.enable_mixed_precision_graph_rewrite`. 
By adding the wrapper of `enable_mixed_precision_graph_rewrite` to the `dnn_optimizer` in the [census wide-n-deep model](https://github.com/baidu-research/tensorflow-allreduce/blob/master/tensorflow/examples/learn/wide_n_deep_tutorial.py#L147), I find the tensorflow internal error of `ValueError: Tensor("current_loss_scale/Read/ReadVariableOp:0", shape=(), dtype=float32) must be from the same graph as Tensor("head/weighted_loss/Sum:0", shape=(), dtype=float32)`

```
    m = tf.estimator.DNNLinearCombinedClassifier(
        model_dir=model_dir,
        linear_feature_columns=crossed_columns,
        dnn_feature_columns=deep_columns,
        dnn_hidden_units=[100, 50])
```
is changed to
```
    m = tf.estimator.DNNLinearCombinedClassifier(
        model_dir=model_dir,
        linear_feature_columns=crossed_columns,
        dnn_feature_columns=deep_columns,
        dnn_optimizer=tf.train.experimental.enable_mixed_precision_graph_rewrite(
              tf.compat.v1.train.GradientDescentOptimizer(0.05)),
        dnn_hidden_units=[100, 50])
```
Maybe it's a compatible error of estimator and enable_mixed_precision_graph_rewrite, please give some help, thanks.

**Comment by Saduf2019** [comment]
@sjtusmartboy 
Could you please provide with simple stand alone code for us to replicate and analyse, if possible you could share a colab gist.

**Comment by sjtusmartboy** [comment]
@Saduf2019 

[wide_deep.zip](https://github.com/tensorflow/tensorflow/files/4346521/wide_deep.zip)

Please just unzip the wide_deep.zip and run the command of `bash run.sh` in the environment of python 2.7.

If the code line of `dnn_optimizer=tf.train.experimental.enable_mixed_precision_graph_rewrite(tf.compat.v1.train.GradientDescentOptimizer(0.05)),
` is deleted, the whole project runs normally. 

If not, the project has the error of `ValueError: Tensor("current_loss_scale/Read/ReadVariableOp:0", shape=(), dtype=float32) must be from the same graph as Tensor("head/weighted_loss/Sum:0", shape=(), dtype=float32).`

**Comment by sjtusmartboy** [comment]
Another question is that how much gpu memory is required to train a model with N parameters if it is trained with Adagrad?@Saduf2019

**Comment by sjtusmartboy** [comment]
@reedwm hi, is there any update on this issue?

**Comment by zhuzilin** [comment]
I encounter the same issue with keras and all tf.compat.v1.train optimizers in tf2.0. One walk around is to use the counterpart in tf.keras.optmizers.

**Comment by reedwm** [comment]
@sjtusmartboy, can you give a smaller example to reproduce? Alternatively, it looks like you branched off from the official models, so can you point me to a fork of the official models on GitHub that reproduces the issue? I am hesitant to run a lot of code in a zip file without looking through all the code first.

**Comment by sjtusmartboy** [comment]
@reedwm I have simplified the code in one file(wide_deep.py)from the original [code](https://github.com/tensorflow/models/tree/d0600d40d02bd927caf97a30895dc482db5d01ab/official/r1/wide_deep) as possible as I can. The model is just wide-deep model with linear column and dnn column. I think it's the problem of using the `enable_mixed_precision_graph_rewrite` for the parameter of `dnn_optimizer` in the `tf.estimator.DNNLinearCombinedClassifier`. So maybe the provided code above is the most simple code that can be submit.  Please recheck, thanks.

**Comment by sjtusmartboy** [comment]
@zhuzilin Thank you for your advice. Now I've change the code to 

`

    m = tf.estimator.DNNLinearCombinedClassifier(
        model_dir=model_dir,
        linear_feature_columns=crossed_columns,
        dnn_feature_columns=deep_columns,
        dnn_optimizer=tf.train.experimental.enable_mixed_precision_graph_rewrite(
                tf.keras.optimizers.SGD(0.05)),
        dnn_hidden_units=[100, 50])

Unfortunately there seems to have some other bugs, `ValueError: The given object is not an Optimizer instance. Given:<tensorflow.python.keras.mixed_precision.experimental.loss_scale_optimizer.LossScaleOptimizer object at 0x7fcbc1943350>`

**Comment by zhuzilin** [comment]
@sjtusmartboy It seems like estimator in 1.14 cannot use v2 optimizer yet. if it's not convenient to upgrade tf, maybe my fix in this pr would help #37965

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
